app/routes/erddap/buoy.js

```diff
@@ -26,44 +33,59 @@ const { getMultiBuoyGeoJsonData, getBuoysCoordinates } = require('../../clients/
  *        in: query
  *        description: Start Time
  *        required: true
- *        type: datetime
+ *        type: string
  *      - name: end
  *        in: query
  *        description: End Time
  *        required: true
- *        type: datetime
+ *        type: string
+ *      - name: numPoints
+ *        in: query
+ *        description: max number of points to return
+ *        required: false
+ *        type: integer
  *     responses:
  *       200:
  *         description: Success! New content is now available.
- * 
+ *
  */
 
-// Ex:  http://localhost:3004/erddap/buoy?datasetId=combined_e784_bee5_492e&ids=bid2&variable=WaterTempSurface&start=2010-07-01T12:00:00Z&end=2010-07-05T12:00:00Z
+// Ex:  http://localhost:3004/erddap/buoy/query?datasetId=combined_e784_bee5_492e&ids=bid2,bid3&variable=WaterTempSurface&start=2010-07-01T12:00:00Z&end=2010-07-05T12:00:00Z
 
-router.get('/buoy', (req, res) => { 
+router.get('/query', (req, res) => {
   const ids = req.query.ids.split(',');
+  const variable = req.query.variable;
   const payload = {
-    ids, 
-    datasetId: req.query.datasetId, 
-    variable: req.query.variable,
-    start: req.query.start, 
-    end: req.query.end
+    ids,
+    datasetId: req.query.datasetId,
+    variable,
+    start: req.query.start,
+    end: req.query.end,
   };
+  const numPoints = req.query.numPoints || 1000;
 
-  return Promise.all(getMultiBuoyGeoJsonData(payload)).then(
-    (response) => {
-      console.log(response);
-      const data = response.map((datum) => {
-        return datum.data.features.map((feature) => {
+  return Promise.all(getMultiBuoyGeoJsonData(payload)).then((response) => {
+    const data = response.map((datum) => {
+      if (datum.hasOwnProperty('data')) {
+        let processed = datum.data.features.map((feature) => {
```

Comment:
should this data re-mapping be moved into the `getMultiBuoyGeoJSONData`?  That way if we use it outside of `/query` there's consistent keys?

Reply:
good point